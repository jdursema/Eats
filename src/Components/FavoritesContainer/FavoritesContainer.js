import React from 'react';
import { connect } from 'react-redux';
import { postAddFavorite, postDeleteFavorite } from '../../Actions';
import Card from '../Card/Card';
import PropTypes from 'prop-types';
import CardContainer from '../CardContainer/CardContainer';


const FavoritesContainer = props => {
  return (
    <div>
      <h1>Your favorite restaurants</h1>
      <CardContainer
        cards = {props.favorites}
        handleAddFav= {props.handleAddFav}
        handleDeleteFav = {props.handleDeleteFav}
        user = {props.user}
        history = {props.history} 
        favorites = {props.favorites}
      />
    </div>
  );
  
};

const mapDispatchToProps = dispatch => ({
  handleAddFav: (cardData, user) => 
    dispatch(postAddFavorite(cardData, user)),
  handleDeleteFav: (cardData, user) =>
    dispatch(postDeleteFavorite(cardData, user))
});

const mapStateToProps = state => ({
  favorites: state.favorites,
  user: state.user
});

export default connect(mapStateToProps, mapDispatchToProps)(FavoritesContainer);

FavoritesContainer.propTypes = {
  cards: PropTypes.array,
  favorites: PropTypes.array
};