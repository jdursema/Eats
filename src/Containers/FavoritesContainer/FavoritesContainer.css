.favorites h1 {
  font-family: 'Raleway', sans-serif;
  color: #0F2130; }
