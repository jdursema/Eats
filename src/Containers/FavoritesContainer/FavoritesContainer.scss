.favorites{
  display: flex;
  flex-direction: column;
  h1{
  color: #0F2130;
  font-family: 'Raleway', sans-serif;
  align-content: center;
  margin: 2%;
  }
}