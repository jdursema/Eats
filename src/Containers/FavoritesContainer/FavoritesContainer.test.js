import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { FavoritesContainer } from './FavoritesContainer';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Favorites', () => {
  it('should match the snapshot', ()=> {

    const renderedFavoritesContainer = shallow(<FavoritesContainer />);

    expect(renderedFavoritesContainer).toMatchSnapshot();
  });
});