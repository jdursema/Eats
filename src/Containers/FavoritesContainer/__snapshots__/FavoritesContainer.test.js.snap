// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Places you already love...
</h1>,
        <CardContainer
          cards={undefined}
          favorites={undefined}
          handleAddFav={undefined}
          handleDeleteFav={undefined}
          history={undefined}
          user={undefined}
/>,
      ],
      "className": "favorites",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Places you already love...",
        },
        "ref": null,
        "rendered": "Places you already love...",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cards": undefined,
          "favorites": undefined,
          "handleAddFav": undefined,
          "handleDeleteFav": undefined,
          "history": undefined,
          "user": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Places you already love...
</h1>,
          <CardContainer
            cards={undefined}
            favorites={undefined}
            handleAddFav={undefined}
            handleDeleteFav={undefined}
            history={undefined}
            user={undefined}
/>,
        ],
        "className": "favorites",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Places you already love...",
          },
          "ref": null,
          "rendered": "Places you already love...",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cards": undefined,
            "favorites": undefined,
            "handleAddFav": undefined,
            "handleDeleteFav": undefined,
            "history": undefined,
            "user": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
